Mimir.Worker/ActionHandler/RuneSlotStateHandler.cs

```diff
@@ -11,69 +10,119 @@
 
 namespace Mimir.Worker.ActionHandler;
 
-public class RuneSlotStateHandler(IStateService stateService, MongoDbService store)
-    : BaseActionHandler(
+public class RuneSlotStateHandler(IStateService stateService, MongoDbService store) :
+    BaseActionHandler(
         stateService,
         store,
-        "^(battle_arena[0-9]*|event_dungeon_battle[0-9]*|join_arena[0-9]*|hack_and_slash[0-9]*|hack_and_slash_sweep[0-9]*|raid[0-9]*|unlock_rune_slot[0-9]*)$",
-        Log.ForContext<RuneSlotStateHandler>()
-    )
+        "^(battle_arena[0-9]*|event_dungeon_battle[0-9]*|hack_and_slash[0-9]*|hack_and_slash_sweep[0-9]*|join_arena[0-9]*|raid[0-9]*|unlock_rune_slot[0-9]*)$",
+        Log.ForContext<RuneSlotStateHandler>())
 {
+    private static readonly BattleArena BattleArena = new();
+    private static readonly EventDungeonBattle EventDungeonBattle = new();
+    private static readonly HackAndSlash HackAndSlash = new();
+    private static readonly HackAndSlashSweep HackAndSlashSweep = new();
+    private static readonly JoinArena JoinArena = new();
+    private static readonly Raid Raid = new();
+    private static readonly UnlockRuneSlot UnlockRuneSlot = new();
```

Comment:
I think that it is a bad idea to use shared action instances. My feeling is that LoadPlainValue will be fine for parsing the key fields, but I'm concerned about the possible side effects of sharing the same action.

After researching this case, I found one case. (there may be more.)

The following lines are parts of `HackAndSlash.LoadPlainValue`:

```csharp
Costumes = ((List)plainValue["costumes"]).Select(e => e.ToGuid()).ToList();
Equipments = ((List)plainValue["equipments"]).Select(e => e.ToGuid()).ToList();
Foods = ((List)plainValue["foods"]).Select(e => e.ToGuid()).ToList();
RuneInfos = plainValue["r"].ToList(x => new RuneSlotInfo((List)x));
WorldId = plainValue["worldId"].ToInteger();
StageId = plainValue["stageId"].ToInteger();
if (plainValue.ContainsKey("stageBuffId"))
{
    StageBuffId = plainValue["stageBuffId"].ToNullableInteger();
}
AvatarAddress = plainValue["avatarAddress"].ToAddress();
TotalPlayCount = plainValue["totalPlayCount"].ToInteger();
ApStoneCount = plainValue["apStoneCount"].ToInteger();
```

Let's focus on the parts to set `StageBuffId`.

```csharp
if (plainValue.ContainsKey("stageBuffId"))
{
    StageBuffId = plainValue["stageBuffId"].ToNullableInteger();
}
```

When you call `HackAndSlash.LoadPlainValue` with the `plainValue` including `StageBuffId=4` first time, the call will set `StageBuffId` field to `4`.

But, when you call `HackAndSlash.LoadPlainValue` with the `plainValue` without `stageBuffId` as second time, the call will not set `StageBuffId` field but the `StageBuffId` is still `4`.

Like this example, I think such usage can pollute related data and it should be recommended.

Reply:
Great catch! Thanks!